cpg_workflows/jobs/rd_combiner/vqsr.py

```diff
@@ -176,3 +177,136 @@ def train_vqsr_snps(sites_only_vcf: str, snp_model: str, job_attrs: dict):
 
     get_batch().write_output(snp_recalibrator_j.model_file, snp_model)
     return snp_recalibrator_j
+
+
+def train_vqsr_snp_tranches(
+    manifest_file: Path,
+    snp_model_path: str,
+    output_path: str,
+    temp_path: Path,
+    job_attrs: dict,
+) -> list[Job]:
+    """
+    train vqsr tranches for SNPs, in a scattered manner
+    Args:
+        manifest_file (Path): Path object to the manifest file
+        snp_model_path (str): path to the SNP model trained in the previous stage
+        output_path (str): path to write the tranches to
+        temp_path (Path): path to the temp directory
+        job_attrs (dict): job attributes
+
+    Returns:
+        all jobs required to get where we're going
+    """
+
+    vcf_resources = get_all_fragments_from_manifest(manifest_file)
+
+    fragment_count = len(vcf_resources)
+    snps_recal_paths = [temp_path / f'snp_{i}.recal' for i in range(fragment_count)]
+    snps_tranches_paths = [temp_path / f'snp_{i}.tranches' for i in range(fragment_count)]
+
+    snp_model_in_batch = get_batch().read_input(snp_model_path)
+
+    resources = get_localised_resources_for_vqsr()
+
+    # the list of all jobs (per-fragment, and the summary)
+    scatter_jobs: list[Job] = []
+
+    # to hold the resulting parts, as Resources
+    snp_tranche_fragments: list[ResourceFile] = []
+
+    # if we start this as -1, we can increment at the start of the loop, making the index counting easier to track
+    vcf_counter = -1
+
+    # iterate over all fragments, but in chunks of FRAGMENTS_PER_JOB
+    for chunk_counter, fragment_chunk in enumerate(chunks(vcf_resources, TRAINING_PER_JOB)):
+        # NB this VQSR training stage is scattered, and we have a high number of very small VCF fragments
+        # 99.5% of the time and cost of this task was pulling the docker image and loading reference data
```

Comment:
Are these time estimates from the tiny test cohort, or did you look at the equivalent jobs from the previous pipeline? It will be interesting to see how this scales.

Reply:
These were the timings from the test cohort. From the genome runs in `main` the ratio was a bit different... maybe 2 mins reading references to 40 seconds running the process.

Reply:
In seqr-loader (lower scatter count at ~1500) the balance was ~2 mins to run the input step, ~40 seconds of work
https://batch.hail.populationgenomics.org.au/batches/530803/jobs/6177. 140GB storage provisioned but only 20GB used, almost entirely references